projects/igniteui-angular/src/lib/directives/for-of/for_of.directive.ts

```diff
@@ -410,15 +410,23 @@ export class IgxForOfDirective<T> implements OnInit, OnChanges, DoCheck, OnDestr
             return;
         }
         // this.state.startIndex = index;
+        const containerSize = parseInt(this.igxForContainerSize, 10);
+        const isPrevItem = index < this.state.startIndex;
         if (this.igxForScrollOrientation === 'horizontal') {
-            this.hScroll.scrollLeft = this.sizesCache[index];
+            this.hScroll.scrollLeft = isPrevItem ?
+            this.sizesCache[index] :
+            this.sizesCache[index] - containerSize + this.igxForOf[index].width;
```

Comment:
I recall that `this.igxForOf[index].width` returns a string. Maybe a `parseInt`?
Also is there a case when this whole expression can be less than 0 in the case when it's not a prevItem? Maybe wrap it in `Math.abs` or some other check?